### ISSUE TITLE
Not an issue 

### ISSUE THREAD

**Issue by guavuavcycayfayqvqu** [OttoDIY/OttoDIYLib#70]
I asked for adding support for otto ninja using esp32 but they said no good servo library for esp32. There exists https://github.com/pvanallen/VarSpeedPython and https://github.com/RoboticsBrno/ServoESP32    can you please port them for esp32.

**Comment by sfranzyshen** [comment]
Hi @guavuavcycayfayqvqu 

> I asked for adding support for otto ninja using esp32 but they said no good servo library for esp32. There exists https://github.com/pvanallen/VarSpeedPython and https://github.com/RoboticsBrno/ServoESP32 can you please port them for esp32.

We are currently working on a rework of the arduino library to support several platforms ... esp32 being just one of them. We will also be unifying the code for the various models (biped, wheels, and ninja) ... The project outline can be found here 
https://github.com/OttoDIY/OttoDIYLib/tree/devel-13.1.0  ... work is slow and things won't be available for awhile ...

I am not sure who (or where) you asked for help but there is a facebook group and discord server with many users that can help you ... here 

https://discord.gg/BhxefDPz
https://www.facebook.com/groups/ottodiy

I will leave it up to you to decide on if and when to close this issue 

~peace

**Comment by guavuavcycayfayqvqu** [comment]
@cparrapa told that there is no good servo library so i tried to make one i already mentioned link in other issue and requested pull.

**Comment by guavuavcycayfayqvqu** [comment]
Can you check out it please.

**Comment by guavuavcycayfayqvqu** [comment]
Maybe esp32 support take a while team has to fix library, make custom pcb and maybe needed to modify 3d design 😔.

**Comment by sfranzyshen** [comment]
> team has to fix library

We are working on it ... https://github.com/OttoDIY/OttoDIYLib/tree/devel-13.1.0

> make custom pcb and 

I agree! but instead of a custom mcu board ... we need a custom power board more ...

> maybe needed to modify 3d design 😔.

We are working on it too ... https://www.tinkercad.com/users/9Wbn1uShxa2?size=small

**Comment by sfranzyshen** [comment]
> Can you check out it please.

yes, I will review your changes ... but after merging ninja into the main Otto code ... the OttoNinja package will be retired ... so any changes or patches will only be applied to the OttoDIYLib package (repo) ...

**Comment by cparrapa** [comment]
Hi guys @guavuavcycayfayqvqu  @sfranzyshen 

I labeled this as enhancement 

I meant there is no good ESP32 library that will be automatically ported, specially with the oscillator dependency it needs to be tested with the hardware.

Guys instead of the custom board, the new HP Otto board already works for this, comes with the right power and the compatibility is kept across the projects, including the modularity.  https://hprobots.com/otto-robot/product/ it will be faster approach to implement the ESP32 support.

**Comment by guavuavcycayfayqvqu** [comment]
Yes it's great. But in my country they don't ship and it's fun to make by myself 😁.

**Comment by guavuavcycayfayqvqu** [comment]
Hp robots is making custom pcb and using own library but core is otto they
are doing business with open source otto. Isn't it against rules? They
didn't made esp32 pcb layout public.

**Comment by cparrapa** [comment]
> Yes it's great. But in my country they don't ship and it's fun to make by myself 😁.

If you mean for direct sales you can contact them and they will arrange which country are you from?
HP Robots and Otto DIY are under the same roof. 🤗
